lib/active_merchant/billing/gateways/adyen.rb

```diff
@@ -124,14 +127,30 @@ def add_shopper_interaction(post, payment, options={})
       def add_address(post, options)
         return unless post[:card] && post[:card].kind_of?(Hash)
         if (address = options[:billing_address] || options[:address]) && address[:country]
-          post[:card][:billingAddress] = {}
-          post[:card][:billingAddress][:street] = address[:address1] || 'N/A'
-          post[:card][:billingAddress][:houseNumberOrName] = address[:address2] || 'N/A'
-          post[:card][:billingAddress][:postalCode] = address[:zip] if address[:zip]
-          post[:card][:billingAddress][:city] = address[:city] || 'N/A'
-          post[:card][:billingAddress][:stateOrProvince] = address[:state] if address[:state]
-          post[:card][:billingAddress][:country] = address[:country] if address[:country]
+          post[:billingAddress] = {}
+          post[:billingAddress][:street] = address[:address1] || 'N/A'
+          post[:billingAddress][:houseNumberOrName] = address[:address2] || 'N/A'
+          post[:billingAddress][:postalCode] = address[:zip] if address[:zip]
+          post[:billingAddress][:city] = address[:city] || 'N/A'
+          post[:billingAddress][:stateOrProvince] = address[:state] if address[:state]
+          post[:billingAddress][:country] = address[:country] if address[:country]
```

Comment:
This conditional seems repetitive as you already have that conditional here: https://github.com/activemerchant/active_merchant/pull/2818/files#diff-190c63e5fc794acdcbe09e98ab586881R129